specification/transport_layer/can.tex

```diff
@@ -23,235 +23,252 @@ \subsection{CAN ID structure}
-\cline{2-4} \cline{7-10} \cline{23-29}
-
- &
-\multicolumn{3}{c|}{Priority}
- &
- &
- &
-\multicolumn{16}{c|}{} &
-\multicolumn{7}{c|}{Source Node ID} &
-\\
-
- &
-\multicolumn{3}{c|}{[0, 7]} &
-0 &
-0 &
-\multicolumn{4}{c}{} &
-\multicolumn{12}{c|}{[0, 65535]} &
-\multicolumn{7}{c|}{[1, 127]} & 1 \\
-\hline
-
-\textbf{CAN ID Bit} &
-28 & 27 & 26 & 25 & 24 & 23 & 22 & 21 & 20 & 19 & 18 & 17 & 16 & 15 &
-14 & 13 & 12 & 11 & 10 &  9 &  8 &  7 &  6 &  5 &  4 &  3 &  2 &  1 &  0 \\
-\hline
-
-\multicolumn{30}{c}{} \\ \hline %% Table seperator
-
-\multirow{3}{*}{\textbf{Service}} &
-\multicolumn{4}{c|}{service/!message} &
-\multicolumn{5}{c|}{request/!response} &
-\multicolumn{5}{c|}{} &
-\multicolumn{7}{c|}{\multirow{2}{*}{Destination Node ID}} &
-\multicolumn{8}{c|}{Protocol Version} \\
-\cline{2-4} \cline{7-10} \cline{23-29}
-
- &
-\multicolumn{3}{c|}{Priority} &
-\cellcolor{red} &
- &
-\multicolumn{9}{c|}{Service ID} &
-\multicolumn{7}{c|}{} &
-\multicolumn{7}{c|}{Source Node ID} &  \\
- &
-\multicolumn{3}{c|}{[0, 7]} &
-1 &
-0|1 &
-\multicolumn{9}{c|}{[0, 511]} &
-\multicolumn{7}{c|}{[1, 127]} &
-\multicolumn{7}{c|}{[1, 127]} & 1 \\
-\hline
-
-\textbf{CAN ID Bit} &
-28 & 27 & 26 & 25 & 24 & 23 & 22 & 21 & 20 & 19 & 18 & 17 & 16 & 15 &
-14 & 13 & 12 & 11 & 10 &  9 &  8 &  7 &  6 &  5 &  4 &  3 &  2 &  1 &  0 \\
-\hline
-
-\end{tabular}
-}
-
-\caption{CAN ID structure}\label{fig:can_id_structure}
+    \centering
+    \resizebox{\textwidth}{!}{
+        \footnotesize
+        \begin{tabular}{|l|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|} \hline
+            %
+            % Message transfer
+            %
+            \multirow{2}{*}{\textbf{Message}} &
+            \multicolumn{4}{c|}{Service, not message} &
+            \multicolumn{5}{c|}{Anonymous message} &
+            \multicolumn{12}{c|}{\multirow{2}{*}{Subject ID}} &
+            \multicolumn{8}{c|}{Protocol version}
+            \\\cline{2-4} \cline{7-10} \cline{23-29}
+
+            &
+            \multicolumn{3}{c|}{Priority}
+            &
+            &
+            &
+            \multicolumn{16}{c|}{} &
+            \multicolumn{7}{c|}{Source node ID} &
+            \\
+
+            \textbf{Values} &
+            \multicolumn{3}{c|}{$[0, 7]$} &
+            $0$ &
+            $0$ &
+            \multicolumn{4}{c}{} &
+            \multicolumn{12}{c|}{$[0, 65535]$} &
+            \multicolumn{7}{c|}{$[0, 127]$} &
+            $1$
+            \\\hline
+
+            \textbf{CAN ID bit} &
+            28 & 27 & 26 & 25 & 24 & 23 & 22 & 21 & 20 & 19 & 18 & 17 & 16 & 15 &
+            14 & 13 & 12 & 11 & 10 &  9 &  8 &  7 &  6 &  5 &  4 &  3 &  2 &  1 &  0
+            \\\hline
+
+            \textbf{CAN ID byte} &
+            \multicolumn{5}{c|}{3} & \multicolumn{8}{c|}{2} & \multicolumn{8}{c|}{1} & \multicolumn{8}{c|}{0}
+            \\\hline
+
+            \multicolumn{30}{c}{} \\ \hline % Table separator
+
+            %
+            % Anonymous message transfer
```

Comment:
Should we combine the anonymous and normal message diagrams into one? (I find myself comparing each bit looking for what is different which is a little confusing) Basically we'd be saying there are two types; messages and service transfers, but that the message type comes in two flavors; regular and anonymous.

Just a thought. Not sure if this is a good idea or not.

Reply:
Here's a counter-thought: should we split request and response for extra clarity instead?

Reply:
I'd rather see the message types combined but if not then your suggestion is good as well.

Reply:
:+1: for the status quo.

I think anonymous is different enough for splitting and request/response is similar enough to keep together. 

I also think that we're down to a question of preference and any of the suggestions will make do.

Reply:
I think it would be a non-issue if we went back to the portrait CAN ID diagram we used to have before https://github.com/UAVCAN/specification/pull/31. Kjetil, are you still vehemently opposed to using portrait diagrams?

Reply:
I just find the horizontal diagrams much more readable. I honestly think by listing the bits vertically it would take me 2-3 times as long to wrap my head around it.

I think both 2, 3 or 4 horizontal figures would be more readable.

Reply:
Okay, I have collapsed this stuff into just two diagrams as Scott suggested. I found this to be even more sensible if we renamed "payload discriminator" into "source node ID stub", which I just did. Please review.

Reply:
👍 but (because I like green bike sheds) a "stub" is a remnant or abbreviation of a full thing. The anonymous identifier is neither since the sender has not node ID. Perhaps calling it the "source node dummy ID" or "source node psedo-ID" would be more accurate?

Reply:
Replaced with pseudo-ID.